### PR TITLE
cppcheck cleanups

### PR DESCRIPTION
**Pull request by mr-c**

### CODE REVIEW

**Comment by mr-c** [comment]
- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Is the Copyright year up to date?

**Comment by ctb** [comment]
This doesn't compile for me on Mac OS X with a syntax error --

```
lib/hashtable.hh:245:49: error: expected ';' at end of declaration list
    const Hashtable& operator=(const Hashtable&)
                                                ^
                                                ;
lib/hashtable.hh:197:9: error: use of undeclared identifier 'partition'
        partition = new SubsetPartition(this);
        ^
lib/hashtable.hh:205:16: error: use of undeclared identifier 'partition'; did
      you mean 'std::partition'?
        delete partition;
               ^~~~~~~~~
               std::partition
/usr/include/c++/4.2.1/bits/stl_algo.h:2097:5: note: 'std::partition' declared
      here
    partition(_ForwardIterator __first, _ForwardIterator __last,
    ^
```

**Comment by mr-c** [comment]
Jenkins, test this please

**Comment by mr-c** [comment]
@ctb shall I merge?

File: khmer/_khmer.cc [hunk 1]
```diff
@@ -4729,9 +4728,9 @@ static PyObject * forward_hash(PyObject * self, PyObject * args)
         return NULL;
     }
 
-    PyObject * hash;
     try {
-        hash = PyLong_FromUnsignedLongLong(_hash(kmer, ksize));
+        PyObject * hash;
```
Comment (ctb) [inline khmer/_khmer.cc]: inconsistent indentation

**Comment by ctb** [comment]
Looks good to me - go ahead and merge (altho maybe fix that indentation).

Question: why the switch from `x++` to `++x` in all the for loops? I don't see the difference :)

**Comment by mr-c** [comment]
`(performance) Prefer prefix ++/-- operators for non-primitive types.`

On Mon, Jul 6, 2015 at 9:57 PM C. Titus Brown <notifications@github.com>
wrote:

> Looks good to me - go ahead and merge (altho maybe fix that indentation).
>
> Question: why the switch from x++ to ++x in all the for loops? I don't
> see the difference :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1144#issuecomment-118995884>.
>
-- 
Michael R. Crusoe: Programmer & Bioinformatician crusoe@ucdavis.edu
The lab for Data Intensive Biology; University of California, Davis
https://impactstory.org/MichaelRCrusoe http://twitter.com/biocrusoe

**Comment by ctb** [comment]
On Mon, Jul 06, 2015 at 02:15:22PM -0700, Michael R. Crusoe wrote:
> `(performance) Prefer prefix ++/-- operators for non-primitive types.`

huh. ok:

https://en.wikibooks.org/wiki/Optimizing_C%2B%2B/Writing_efficient_code/Constructions_and_destructions

learn something every day!!

--titus

**Pull request merged**